test/verify/storagelib.py

```diff
@@ -311,12 +314,37 @@ def dialog_check(self, expect):
                 return False
         return True
 
+    def dialog_set_vals(self, values):
+        # Sometimes a certain field needs to be set before other
+        # fields come into existence and thus the order matter that we
+        # set the fields in.  The tests however just give us a
+        # unordered 'dict'.  Instead of changing the tests, we figure
+        # out the right order dynamically here by just setting what we
+        # can and then starting over.  As long as we make progress in
+        # each iteration, everything is good.
+        progress = False
+        all = True
+        for f in values:
+            try:
+                self.dialog_set_val(f, values[f])
+                progress = True
+            except:
```

Comment:
Please no blanket exception catching. This might also be something really unexpected like `AttributeError`.

Reply:
I agree, but I think this might be the exception (hah!) to the rule.  If the loop doesn't make any progress due to all fields raising an exception, the code will set them all again (below) and then we get a meaningful backtrace.

Reply:
Wouldn't the more meaningful backtrace for syntax errors, variable name typos, and the like be to crash immediately right here, instead of some completely unrelated synthesized exception further down?

Reply:
Now I get what you are saying!  And this code turns out to be quite broken, actually.  It was never triggered in the tests.  (It doesn't work to start over with the full set of field values, that makes it look as if we make progress on every attempt.)

So, fixed.